apps/indexer/lib/indexer/fetcher/on_demand/token_instance_metadata_refetch.ex

```diff
@@ -39,60 +38,63 @@ defmodule Indexer.Fetcher.OnDemand.TokenInstanceMetadataRefetch do
            {:retry,
             CountersHelper.current_time() - updated_at_ms >
               threshold(retries_number)} do
-      fetch_and_broadcast_metadata(token_instance, state)
+      fetch_and_broadcast_metadata(token_instance)
     else
       {:retries_number, nil} ->
-        fetch_and_broadcast_metadata(token_instance, state)
+        fetch_and_broadcast_metadata(token_instance)
 
       {:retry, false} ->
+        Publisher.broadcast(
+          %{
+            not_fetched_token_instance_metadata: [
+              to_string(token_instance.token_contract_address_hash),
+              NFT.prepare_token_id(token_instance.token_id),
+              "retry_cooldown"
+            ]
+          },
+          :on_demand
+        )
+
         :ok
     end
   end
 
-  defp fetch_and_broadcast_metadata(token_instance, _state) do
-    token_id = NFT.prepare_token_id(token_instance.token_id)
-    contract_address_hash_string = to_string(token_instance.token_contract_address_hash)
-
-    request =
-      NFT.prepare_request(
-        token_instance.token.type,
-        contract_address_hash_string,
-        token_id,
-        false
-      )
-
-    result =
-      case Reader.query_contracts([request], NFT.erc_721_1155_abi(), [], false) do
-        [ok: [uri]] ->
-          {:ok, [uri]}
-
-        _ ->
-          nil
-      end
-
-    with {:empty_result, false} <- {:empty_result, is_nil(result)},
-         {:fetched_metadata, {:ok, %{metadata: metadata}}} <-
-           {:fetched_metadata,
-            result
-            |> MetadataRetriever.fetch_json(token_id, nil, false)
-            |> MetadataRetriever.parse_fetch_json_response()} do
-      TokenInstance.set_metadata(token_instance, metadata)
-
-      Publisher.broadcast(
-        %{fetched_token_instance_metadata: [to_string(token_instance.token_contract_address_hash), token_id, metadata]},
-        :on_demand
-      )
-
-      Queue.process_new_instances([%TokenInstance{token_instance | metadata: metadata}])
-    else
-      {:empty_result, true} ->
-        :ok
+  defp fetch_and_broadcast_metadata(token_instance) do
+    case Helper.batch_prepare_instances_insert_params([
+           %{contract_address_hash: token_instance.token_contract_address_hash, token_id: token_instance.token_id}
+         ]) do
+      [%{error: nil, metadata: metadata} = result] ->
+        TokenInstance.set_metadata(token_instance, result)
+
+        Publisher.broadcast(
+          %{
+            fetched_token_instance_metadata: [
+              to_string(token_instance.token_contract_address_hash),
+              NFT.prepare_token_id(token_instance.token_id),
+              metadata
+            ]
+          },
+          :on_demand
+        )
+
+        Queue.process_new_instances([%TokenInstance{token_instance | metadata: metadata}])
```

Comment:
Nitpick: define `token_contract_address_hash` and `token_id` right in the function arguments to replace long `token_instance.token_contract_address_hash` to `token_contract_address_hash` 
and
`token_instance.token_id` to `token_id`

```suggestion
  defp fetch_and_broadcast_metadata(
         %{token_id: token_id, token_contract_address_hash: token_contract_address_hash} = token_instance
       ) do
    case Helper.batch_prepare_instances_insert_params([
           %{contract_address_hash: token_contract_address_hash, token_id: token_id}
         ]) do
      [%{error: nil, metadata: metadata} = result] ->
        TokenInstance.set_metadata(token_instance, result)

        Publisher.broadcast(
          %{
            fetched_token_instance_metadata: [
              to_string(token_contract_address_hash),
              NFT.prepare_token_id(token_id),
              metadata
            ]
          },
          :on_demand
        )

        Queue.process_new_instances([%TokenInstance{token_instance | metadata: metadata}])
```